File: python/scan_image_and_send_email/scan_image_and_send_email/scan_image_and_send_email_stack.py
from aws_cdk import Stack
from aws_cdk import aws_events as events
from aws_cdk import aws_events_targets as targets
from aws_cdk import aws_sns as sns
from aws_cdk import aws_sns_subscriptions as subscriptions
from constructs import Construct


class ScanImageAndSendEmailStack(Stack):

    def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
        super().__init__(scope, construct_id, **kwargs)

        rule = events.Rule(self, "rule",
                           event_pattern=events.EventPattern(
                               source=["aws.ecr"],
                               detail_type=["ECR Image Scan"],
                               detail={
                                   "finding-severity-counts": {
                                       "HIGH": [{
                                           "numeric": [">", 0]
                                       }]
                                   }
                               }
                           )
                           )
        topic = sns.Topic(self, "topic",
                          topic_name="holamundo")
        topic.add_subscription(subscriptions.EmailSubscription(
            email_address="holamundo@some-domain.com"))
        rule.add_target(targets.SnsTopic(topic=topic))

File: typescript/scan_image_and_send_email/lib/scan_image_and_send_email-stack.ts
import { Stack, StackProps } from "aws-cdk-lib";
import * as sns from "aws-cdk-lib/aws-sns";
import * as events from "aws-cdk-lib/aws-events";
import * as targets from "aws-cdk-lib/aws-events-targets";
import * as subscriptions from "aws-cdk-lib/aws-sns-subscriptions";
import { Construct } from "constructs";

export class ScanImageAndSendEmailStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const rule = new events.Rule(this, "ScanImageAndSendEmailQueue", {
      eventPattern: {
        source: ["aws.ecr"],
        detailType: ["ECR Image Scan"],
        detail: {
          "finding-severity-counts": {
            HIGH: [
              {
                numeric: [">", 0],
              },
            ],
          },
        },
      },
    });

    const topic = new sns.Topic(this, "ScanImageAndSendEmailTopic", {
      topicName: "holamundo",
    });
    topic.addSubscription(
      new subscriptions.EmailSubscription("holamundo@some-domain.com")
    );

    rule.addTarget(new targets.SnsTopic(topic));
  }
}
